packages/merge-styles/src/styleToClassName.ts

```diff
@@ -0,0 +1,179 @@
+import { GlobalSettings } from '@uifabric/utilities/lib/GlobalSettings';
+import { rtlifyRules } from './transforms/rtlifyRules';
+import { provideUnits } from './transforms/provideUnits';
+import { prefixRules } from './transforms/prefixRules';
+import { expandQuads } from './transforms/expandQuads';
+import { getVendorSettings } from './getVendorSettings';
+import { Stylesheet } from './Stylesheet';
+import { IStyle } from './IStyle';
+
+const DISPLAY_NAME = 'displayName';
+
+function toRuleEntry(name: string, value: string): string | undefined {
+  if (name === DISPLAY_NAME) {
+    return undefined;
+  }
+
+  // Prefixes like WebKit need a dash infront of them.
+  if (/[A-Z]/.test(name[0])) {
+    name = '-' + name;
+  }
+
+  // Kebab case the rule.
+  let rules: string[] = [
+    name.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase(),
+    value
+  ];
+
+  let ruleEntries: string[] = [];
+
+  // Apply transforms.
+  provideUnits(rules);
+  rtlifyRules(rules);
+  prefixRules(rules);
+  expandQuads(rules);
+
+  // Stringify the rules.
+  for (let i = 0; i < rules.length; i += 2) {
+    ruleEntries.push(rules[i], ':', rules[i + 1], ';');
+  }
+
+  return ruleEntries.join('');
+}
+
+function getDisplayName(rules?: Map<string, IStyle>): string | undefined {
+  let rootStyle: IStyle = rules && rules.get('&');
+
+  // tslint:disable-next-line:no-any
+  return rootStyle && (rootStyle as any).displayName;
+}
+
+/**
+ * Given an array of objects or strings, iterate through each:
+ *  if string, look up Map. Merge map into current map.
+ *  else if object:
+ *    iterate through each prop:
+ *      if is a selector.
+ *
+ *
+ * edge cases; should :active always have prescidence over :hover?
+ * ([':hover': {}, ':active': {}], [':hover': {}])
+ */
+function extractRules(
+  args: IStyle[],
+  rules: Map<string, {}> = new Map<string, {}>(),
+  currentSelector: string = '&'
+): Map<string, {}> {
+  let stylesheet = Stylesheet.getInstance();
+  let currentRules: {} | undefined = rules.get(currentSelector);
+
+  if (!currentRules) {
+    currentRules = {};
+    rules.set(currentSelector, currentRules);
+  }
+
+  for (let i = 0; i < args.length; i++) {
+    const arg: IStyle = args[i];
+
+    // If the arg is a string, we need to look up the class map and merge.
+    if (typeof arg === 'string') {
+      let expandedRules = stylesheet.argsFromClassName(arg);
```

Comment:
I don't really agree with it. Here's why.

`const` is very useful for communicating and enforcing that something should not change. There are times where you do this to avoid mistakes. I look at it as a readonly property. Good usage examples are things such as constant strings at the top of a file, which are reused in a variety of places. It communicates and enforces that these aren't going to change values ever.

But most of the time, you don't care to communicate this. And it doesn't prevent bugs when enforced. It just creates noise. I don't think I've ever hit a scenario where I accidentally assigned a value to something where it wasn't anticipated.

If you have a scenario where there is a legitimate bug caught by overusing const, I'd love to hear it.